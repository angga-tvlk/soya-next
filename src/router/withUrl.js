import React from 'react';
import getDisplayName from '../utils/getDisplayName';
import { urlShape } from '../constants/PropTypes';

export default Component => class extends React.Component {
  static displayName = getDisplayName('WithUrl', Component);

  static contextTypes = {
    url: urlShape.isRequired,
  };

  static propTypes = {
    url: urlShape,
  };

  render() {
    return (
      <Component
        {...this.props}
        url={this.props.url || this.context.url}
      />
    );
  }
};
