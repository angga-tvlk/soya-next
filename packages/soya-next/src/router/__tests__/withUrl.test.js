import React from 'react';
import { shallow } from 'enzyme';
import withUrl from '../withUrl';

describe('withUrl', () => {
  it('should add url to component props', () => {
    const context = {
      url: {
        pathname: '/',
        query: {},
      },
    };
    const Component = () => <div />;
    const ComponentWithUrl = withUrl(Component);
    const wrapper = shallow(<ComponentWithUrl />, { context });
    expect(wrapper.find(Component).props()).toMatchSnapshot();
  });
});
