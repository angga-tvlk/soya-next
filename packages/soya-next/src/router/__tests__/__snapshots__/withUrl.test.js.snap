// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`withUrl should add url to component props 1`] = `
Object {
  "url": Object {
    "pathname": "/",
    "query": Object {},
  },
}
`;
